main.go

```diff
@@ -174,6 +175,30 @@ func setupCreateOrderRoute(r *gin.Engine, db *tigris.Database) {
 	})
 }
 
+// Create routes for searching data in a collection
+func setupSearchRoutes[T interface{}](r *gin.Engine, db *tigris.Database, name string) {
+	r.POST(fmt.Sprintf("/%s/search", name), func(c *gin.Context) {
+		coll := tigris.GetCollection[T](db)
+
+		var u search.Request
+		if err := c.Bind(&u); err != nil {
+			return
+		}
+		it, err := coll.Search(c, &u)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+		r := &search.Result[T]{}
+		for it.Next(r) {
+			c.JSON(http.StatusOK, r)
```

Comment:
We need to accumulate results here in a variable and send all at once.
I think c.JSON supposed to be called once, because it sending headers and on second invocation it'll error out.

Other opiton would be to create a server stream https://pkg.go.dev/github.com/gin-gonic/gin#Context.SSEvent but this would be and overkill in the example.

Reply:
I have tested multiple invocations, it outputs json to stdout as expected page by page.

Reply:
I just remember that I saw `"http: superfluous response.WriteHeader call from %s (%s:%d)"`. I don't see it now.
So probably it's ok to leave it like this for now.